大一/数据结构/习题1/test1.cpp

```diff
@@ -0,0 +1,20 @@
+﻿#include <iostream>
+using namespace std;
+
+void hanoi(int n, char a, char c, char b) {
+    if (n == 1) {
+        cout << "移动盘 1 由 " << a << " -> " << c << endl;
+        return ;
+    }
+    hanoi(n - 1, a, b, c);
+    cout << "移动盘 " <<  n  << " 由 " << a << " -> " << c << endl;
+    hanoi(n - 1, b, c, a);
+}
+
+int main() {
+    int n;
+    cout << "请输入圆盘的总数量: ";
+    cin >> n;
```

Comment:
For user input, we should always check its value.

e.g. Must greater than 0 etc.